import config
from mysql.connector import connect, Error
from utils.logging_config import app_logger, error_logger

configure = {
    "user": config.DB_USER,
    "password": config.DB_PASSWORD,
    "host": config.DB_HOST,
    "database": config.DB_NAME,
    "port": config.DB_PORT
}


def get_connection():
    try:
        connection = connect(**configure)
        if connection.is_connected():
            app_logger.info(f"Connection to database {configure['database']} was successfull.")
            return connection
    except Error as e:
        error_logger.error(f"Failed to establish connection to database {configure['database']}: {e}")
        return None
